app/assets/javascripts/dashboard/dashboard.js

```diff
@@ -320,12 +353,23 @@ MinionPoller = {
         break;
     }
 
+    if (State.pendingRemovalMinionId || State.hasPendingStateNode) {
+      actionsHtml = '<a href="#" class="disabled remove-node-link" data-id="' + minion.minion_id + '">Remove</a>';
+    } else {
+      actionsHtml = '<a href="#" class="remove-node-link" data-id="' + minion.minion_id + '">Remove</a>';
```

Comment:
should we open a confirmation modal/dialog before actually performing the call? for such a destructive action IMO it's better to prevent accidental removes

Reply:
Yes, we should do the same as we did for the registries/mirrors.